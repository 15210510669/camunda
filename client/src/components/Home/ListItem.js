/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';
import classnames from 'classnames';

import './ListItem.scss';

export default function ListItem({className, children, onClick}) {
  return (
    <li className={classnames('ListItem', className, {clickable: onClick})} onClick={onClick}>
      <div className="indicator" />
      <ul>{children}</ul>
    </li>
  );
}

ListItem.Section = function ListItemSection({className, ...props}) {
  return <li className={classnames('ListItemSection', className)} {...props} />;
};
