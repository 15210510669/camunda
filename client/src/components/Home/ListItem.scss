.ListItem {
  width: 100%;
  height: 56px;
  border-radius: 4px;
  border: solid 1px #d3d3d3;
  background-color: #ffffff;
  margin-bottom: 10px;
  display: block;

  .indicator {
    width: 8px;
    height: calc(100% + 2px);
    margin-top: -1px;
    margin-left: -1px;
    border-radius: 4px 0 0 4px;
    display: inline-block;
    margin-right: 17px;
  }

  ul {
    display: inline-block;
    list-style-type: none;
    height: 100%;
    vertical-align: top;
  }
}

.ListItemSection {
  height: 100%;
  padding-right: 10px;
  display: inline-block;
  vertical-align: top;
  line-height: 54px;

  font-size: 13px;
  font-weight: normal;
  color: #484c57;

  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}
