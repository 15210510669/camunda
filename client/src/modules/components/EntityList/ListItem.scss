@import 'shared-styles/shared-styles.scss';

.ListItem {
  width: 100%;
  height: 60px;
  border-radius: 3px;
  box-shadow: 0 1px 4px 0 rgba(0, 0, 0, 0.1);
  border: solid 1px $silver-darken-87;
  background-color: #ffffff;
  margin-bottom: 14px;
  list-style: none;

  > a:hover {
    background-color: $silver-lighten-99;
  }

  > a,
  > div {
    display: flex;
    align-items: center;
    height: 100%;
    color: inherit;
    text-decoration: none;

    > .Icon {
      flex: 0 0 24px;
      font-size: 24px;
      margin-left: 15px;
      margin-right: 0;
    }
  }

  .Input[type='checkbox'] {
    margin-left: 20px;
    margin-right: 0;
    flex: 0 0 18px;
    height: 18px;
    visibility: hidden;
  }

  .name {
    margin-left: 19px;
    display: flex;
    flex-direction: column;
    flex-grow: 1;
    padding-right: 10px;
    overflow: hidden;

    .type {
      color: $grey-lighten-50;
      font-size: 12px;
      line-height: 1.5;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    }
    .entity {
      color: $grey-darken-23;
      font-size: 15px;
      line-height: 1.2;
      font-weight: 500;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    }
  }

  .meta {
    color: $grey-lighten-50;
    font-size: 14px;
    flex: 0 0 200px;
    padding-right: 10px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  .warning {
    flex: 0 0 50px;
    color: $red-darken-38;
    text-align: center;
    position: relative;
    height: 18px;

    .Icon {
      margin: 0;
      font-size: 18px;
      vertical-align: initial;
    }

    .Tooltip {
      display: none;
      position: absolute;
      top: 18px;
      left: 50%;
      transform: translateX(-50%);

      .Tooltip__text-bottom {
        width: max-content;
        max-width: 170px;
      }
    }

    &:hover .Tooltip {
      display: block;
    }
  }

  &:not(:hover) .ListItemAction {
    .Button,
    &.Button {
      opacity: 0;

      &.active,
      &.focus-visible {
        opacity: 1;
      }
    }
  }
}
