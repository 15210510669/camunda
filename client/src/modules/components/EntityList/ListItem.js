/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import React from 'react';
import {Link} from 'react-router-dom';
import classnames from 'classnames';

import {Input, Icon, Tooltip} from 'components';

import ListItemAction from './ListItemAction';

import './ListItem.scss';

export default function ListItem({
  isSelected,
  onSelectionChange,
  data: {type, name, link, icon, meta = [], actions, warning},
  hasWarning,
  singleAction,
}) {
  const content = (
    <>
      <Icon type={icon} />
      <div className="name">
        <span className="type">{type}</span>
        <Tooltip content={name} overflowOnly>
          <span className="entity">{name}</span>
        </Tooltip>
      </div>
      {meta.map((content, idx) => (
        <Tooltip key={idx} content={content} overflowOnly>
          <div className="meta">{content}</div>
        </Tooltip>
      ))}
      {hasWarning && (
        <div className="warning">
          {warning && (
            <Tooltip content={warning} theme="dark" delay={0}>
              <Icon type="error" size="18" />
            </Tooltip>
          )}
        </div>
      )}
      <ListItemAction actions={actions} singleAction={singleAction} />
    </>
  );

  return (
    <li
      className={classnames('ListItem', {
        active: isSelected,
        selectable: actions?.length > 0,
      })}
    >
      <Input type="checkbox" checked={isSelected} onChange={onSelectionChange} />
      {link ? <Link to={link}>{content}</Link> : <div>{content}</div>}
    </li>
  );
}
