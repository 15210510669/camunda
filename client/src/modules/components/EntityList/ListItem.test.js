/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import React from 'react';
import {shallow} from 'enzyme';

import ListItem from './ListItem';

it('should match snapshot', () => {
  const node = shallow(
    <ListItem
      data={{
        type: 'Item Type',
        name: 'Test List Entry',
        icon: 'process',
        link: '/report/1',
        meta: ['Column 1', 'Column 2'],
        actions: [{action: jest.fn(), icon: 'delete', text: 'Delete Entry'}],
        warning: 'Warning Text',
      }}
      hasWarning
    />
  );

  expect(node).toMatchSnapshot();
});

it('should not render a Link component if no link is provided', () => {
  const node = shallow(<ListItem data={{}} />);

  expect(node.find('Link')).not.toExist();
});

it('should have a warning column even if no specific warning exists for this ListItem', () => {
  const node = shallow(<ListItem data={{}} hasWarning />);

  expect(node.find('.warning')).toExist();
});

it('should invoke onSelectionChange when checkbox is triggered', () => {
  const spy = jest.fn();
  const node = shallow(<ListItem data={{}} onSelectionChange={spy} />);

  node.find({type: 'checkbox'}).simulate('change', 'test');

  expect(spy).toHaveBeenCalledWith('test');
});
