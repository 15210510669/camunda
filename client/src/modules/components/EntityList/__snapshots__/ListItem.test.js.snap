// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should match snapshot 1`] = `
<li
  className="ListItem list-test"
>
  <div
    className="indicator"
  />
  <ul>
    Some child content
  </ul>
</li>
`;

exports[`ListItem.Section should match snapshot 1`] = `
<li
  className="ListItemSection list-test"
>
  Some child content
</li>
`;
